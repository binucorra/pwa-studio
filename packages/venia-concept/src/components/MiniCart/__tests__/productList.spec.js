import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import ProductList from '../productList';

jest.mock('../product');
jest.mock('../productListFooter');

const baseProps = {
    cart: {
        details: {
            currency: {
                quote_currency_code: 'US'
            }
        }
    }
};

test('renders a list with no items when items are not supplied', () => {
    const tree = createTestInstance(
        <ProductList {...baseProps} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('renders a list of Products when items are supplied', () => {
    const props = {
        ...baseProps,
        items: [{
            item_id: 1,
            name: 'unit test',
            price: 99,
            product_type: 'product type',
            qty: 1,
            quote_id: 1,
            sku: 'sku'
        }]
    };
    const tree = createTestInstance(
        <ProductList {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});