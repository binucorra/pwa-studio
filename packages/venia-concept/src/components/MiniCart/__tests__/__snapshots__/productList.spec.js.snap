// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of Products when items are supplied 1`] = `
<ul>
  ( Product Component Here )
  ( Product Component Here )
</ul>
`;

exports[`renders a list with no items when items are not supplied 1`] = `<ul />`;
